src/client/pages/ConsentsDataPage.tsx

```diff
@@ -12,13 +15,22 @@ export const ConsentsDataPage = () => {
   const profiling = consents.find(
     (consent) => consent.id === Consents.PROFILING,
   );
+  const advertising = consents.find(
+    (consent) => consent.id === Consents.ADVERTISING,
+  );
+
+  const hasCmpConsent = useCmpConsent();
+  const isDigitalSubscriber = useAdFreeCookie();
+  const shouldPersonalisedAdvertisingPermissionRender =
+    hasCmpConsent && !isDigitalSubscriber;
 
   return (
-    <ConsentsData
-      consented={profiling?.consented}
-      description={profiling?.description}
-      name={profiling?.name}
-      id={profiling?.id || Consents.PROFILING}
-    />
+    <>
+      {shouldPersonalisedAdvertisingPermissionRender ? (
+        <ConsentsData profiling={profiling} advertising={advertising} />
+      ) : (
+        <ConsentsData profiling={profiling} />
+      )}
+    </>
   );
```

Comment:
Potentially nitpicky / code golf, but could avoid the use of a fragment here:

```tsx
return shouldPersonalisedAdvertisingPermissionRender ? (
  <ConsentsData profiling={profiling} advertising={advertising} />
) : (
  <ConsentsData profiling={profiling} />
);
```

Or you could even conditionally add the advertising prop somehow instead of having two separate JSX expressions. For example:

```tsx
return (
  <ConsentsData
    profiling={profiling}
    {...(shouldPersonalisedAdvertisingPermissionRender && { advertising })}
  />
);
```

(spreading `false` in an object doesn't do anything)

Or you could even do

```tsx
let props: ComponentProps<typeof ConsentsData> = { profiling };

if (hasCmpConsent && !isDigitalSubscriber) props = { ...props, advertising };

return <ConsentsData {...props} />;
```

Lots of ways to skin this particular cat.

Reply:
I like the middle way, keeping the logic where it renders so you don't have to scroll :) 
But I do not endorse skinning cats and no animals were harmed writing this code.